src/main/java/org/elasticsearch/rest/action/update/RestUpdateAction.java

```diff
@@ -58,6 +58,7 @@ public void handleRequest(final RestRequest request, final RestChannel channel,
         UpdateRequest updateRequest = new UpdateRequest(request.param("index"), request.param("type"), request.param("id"));
         updateRequest.listenerThreaded(false);
         updateRequest.routing(request.param("routing"));
+        updateRequest.parent(request.param("parent")); // order is important, set it after routing, so it will set the routing
```

Comment:
is it really important? I think it should work fine given that UpdateRequest.parent only sets the routing if null?